src/handlers/Handler.ts

```diff
@@ -450,4 +461,343 @@ export class Handler implements IHandler {
+                    );
+                    await roomInteractionStorage.storeInteractionRoomId(this.room.id);
+                    // Store email data with room interaction
+                    await this.persis.updateByAssociation(
+                        new RocketChatAssociationRecord(
+                            RocketChatAssociationModel.ROOM,
+                            this.room.id,
+                        ),
+                        { emailData },
+                        true,
+                    );
+
+                    // Show simplified email ready message with buttons
+                    const block = this.modify.getCreator().getBlockBuilder();
+                    
+                    const messageText = toolCall.function.name === 'summarize-and-send-email' 
+                        ? t(Translations.LLM_SUMMARY_EMAIL_READY_USER, this.language, { 
+                            name: this.sender.name || this.sender.username,
+                            channelName: this.room.displayName || 'Channel',
+                            subject: emailData.subject 
+                          })
+                        : t(Translations.LLM_EMAIL_READY_USER, this.language, { 
+                            name: this.sender.name || this.sender.username, 
+                            subject: emailData.subject 
+                          });
+                    
+                    block.addSectionBlock({
+                        text: block.newMarkdownTextObject(messageText),
+                    });
+
+                    block.addActionsBlock({
+                        elements: [
+                            block.newButtonElement({
+                                actionId: ActionIds.SEND_EMAIL_EDIT_ACTION,
+                                text: block.newPlainTextObject(t(Translations.EMAIL_EDIT_AND_SEND_BUTTON, this.language)),
+                                style: ButtonStyle.PRIMARY,
+                            }),
+                        ],
+                    });
+
+                    resultMessageBuilder.setBlocks(block.getBlocks());
+                    return this.read.getNotifier().notifyUser(this.sender, resultMessageBuilder.getMessage());
+                }
+
+                // Use the ToolExecutorService for all other tools
+                const toolExecutorService = new ToolExecutorService(
+                    this.app,
+                    this.read,
+                    this.modify,
+                    this.http,
+                    this.persis
+                );
+                
+                const result = await toolExecutorService.executeTool(
+                    toolCall,
+                    this.sender,
+                    this.room,
+                    this.triggerId
+                );
+                
+                // Create formatted message showing the tool execution result
+                const block = this.modify.getCreator().getBlockBuilder();
```

Comment:
this whole method is TOO long it should be a lot smaller divide this in different files and handlers